test/CMakeLists.txt

```
@@ -906,3 +906,12 @@ if(MIOPEN_TEST_CONV)
     COMMAND	$<TARGET_FILE:test_conv2d>	--verbose	--input	1	48	7	7	--weights	1	48	5	5	--pads_strides_dilations	0	0	4	4	1	1
 )
 endif()
+
+# REGRESSION TEST for issue #894
+if(MIOPEN_BACKEND_OPENCL AND MIOPEN_TEST_HALF)
+    set(CONV_OCL_REGRESSION_TEST_ENV MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvOclDirectFwd1x1)
+
+    add_custom_test(test_miopen_conv_ocl_half_regression SKIP_UNLESS_ALL ALLOW_HALF
+    COMMAND	${CONV_OCL_REGRESSION_TEST_ENV} $<TARGET_FILE:test_conv2d>  --verbose   --half  --cmode conv    --pmode default --group-count   1 --input	1   16	7	7	--weights	16	16	1	1	--pads_strides_dilations	0	0	1	1	1	1
+)
+endif()
```

Please merge recent develop and then:
```suggestion
    add_custom_test(test_regression_half HALF_ENABLED FLOAT_DISABLED GFX908_ENABLED
    COMMAND MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvOclDirectFwd1x1 ${CONV_OCL_REGRESSION_TEST_ENV} $<TARGET_FILE:test_conv2d>  --verbose   --half  --cmode conv    --pmode default --group-count   1 --input	1   16	7	7	--weights	16	16	1	1	--pads_strides_dilations	0	0	1	1	1	1
)
endif()
```
Later, we'll add here more FP16-specific regression tests (hopefully, not many ;)))

---

Merged develop with ocl only constraint removed

---

- We need to use `HALF_ENABLED` to enable the test for FP16 (`ALLOW_HALF` is not working anymore)
- The test is not enabled for MI100 by default; please add GFX908_ENABLED for that (I think we want this check, right?)
- This `--half` test will be run by both FP16 and FP32 stages; I recommend adding `FLOAT_DISABLED` to disable it for FP32.
- We do not need the enclosing `if(MIOPEN_TEST_HALF)...endif()` construct anymore.

Some internal documentation about new "attributes" of `add_custom_test`: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/81c0e4f98a548710626b75ef324774a174a80f0e/test/CMakeLists.txt#L295-L319

---

see update

---

Thanks.